napalm/base.py

```diff
@@ -364,3 +364,44 @@ def get_interfaces_counters(self):
             }
         """
         raise NotImplementedError
+
+        def show_route(self, destination = ''):
+            """
+            Returns a dictionary of dictionaries containing details of all available routes to a destination.
+            Each inner dictionary contains the following fields:
+                * active_route (boolean)
+                * age (int),
+                * as_path (string)
+                * local_preference (int)
+                * next_hop (str)
+                * outgoing_interface (str)
+                * preference (int)
+                * protocol (int)
+
+            Example:
+                {
+                    "1.0.0.0/24": [
+                        {
+                            "active_route"      : True,
+                            "age"               : 1368451,
+                            "as_path"           : "15169 I",
+                            "local_preference"  : 200,
+                            "next_hop"          : "206.223.116.21",
+                            "outgoing_interface": "ae10.0",
+                            "preference"        : 170,
+                            "protocol"          : "BGP"
```

Comment:
What if it comes from OSPF? Do you leave the BGP attributes empty?  Fine by me, just clarifying.